File: homework3-2-master/app/src/main/java/com/example/homework3__2/FirstFragment.java
package com.example.homework3__2;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;


public class FirstFragment extends Fragment {

    private Button btn_plus;
    private Button btn_minus;
    private TextView number;
    private  Integer count = 0;

    @Override
    public View onCreateView(LayoutInflater inflater,  ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_first, container, false);

    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        btn_plus = view.findViewById(R.id.plus);
        btn_minus = view.findViewById(R.id.minus);
        number = view.findViewById(R.id.number);
        btn_plus.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                count += 1;
                number.setText(count.toString());
            }
        });

        btn_minus.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                count -= 1;
                number.setText(count.toString());
            }
        });

    }
}

File: homework3-5-master/app/src/main/java/com/example/homework3_5/FirstFragment.kt
package com.example.homework3_5

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.example.homework3_5.databinding.FragmentFirstBinding


class FirstFragment : Fragment() {
    private lateinit var binding: FragmentFirstBinding
    private var isCountTen = false

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = FragmentFirstBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        startIncrementingOrDecrementing()
    }

    private fun startIncrementingOrDecrementing() = with(binding) {
        var count = tvCount.text.toString().toInt().plus(1)
        btnIncDec.setOnClickListener {
            when (isCountTen) {
                false -> {
                    when (count == 10) {
                        false -> {
                            tvCount.text = count++.toString()
                        }
                        true -> {
                            isCountTen = true
                            tvCount.text = count--.toString()
                            btnIncDec.text = "-"
                        }
                    }
                }
                true -> {
                    when (count == 0) {
                        false ->
                            tvCount.text = count--.toString()
                        true -> requireActivity().supportFragmentManager.beginTransaction().replace(
                            R.id.fragment_container,
                            SecondFragment()
                        ).addToBackStack(null).commit()
                    }
                }
            }
        }
    }
}